src/service/verifyService.ts

```diff
@@ -0,0 +1,147 @@
+        logger.info(`${member.displayName} joined ${member.guild.name}`);
+        try {
+            const userData = await getUser(config.apiKey, member.id);
+            const guild = await getGuild(member.guild.id);
+            updateRoleAndNickname(userData, guild, member);
+            sendMessage(
+                member,
+                `Successfully verified user ${member.user.tag}!`,
+                Color.GREEN
+            );
+        } catch (e) {
+            if (e instanceof ApiError && e.code === 2) {
+                sendMessage(
+                    member,
+                    `Your discord account is not verified with Final Earth.
+                Please visit [here](https://www.finalearth.com/discord) and follow the instructions.`,
+                    Color.RED
+                );
+            } else {
+                logger.error(e);
+            }
+        }
+    }
+
+    @EventHandler({ event: "guildMemberRemove" })
+    async memberLeft(member: GuildMember | PartialGuildMember) {
+        logger.info(`${member.displayName} left ${member.guild.name}`);
+        const repository = AppDataSource.getRepository(UserRank);
+        const userRank = await removeMembers(member, repository);
+        if (userRank) repository.save(userRank);
+    }
+
+    @EventHandler({ event: "guildDelete" })
+    async guildDelete(guild: Guild) {
+        logger.info(`Bot kicked from guild: ${guild.name}`);
+        const repository = AppDataSource.getRepository(UserRank);
+        const members = await guild.members.fetch();
+        const users = (
+            await Promise.all(
+                members.map((member) => removeMembers(member, repository))
+            )
+        ).filter(isSome);
+
+        if (users.length > 0) repository.save(users);
+    }
+}
+
+export async function updateRoleAndNickname(
+    user: UserData,
+    guild: GuildEntity,
+    member: GuildMember
+): Promise<void> {
+    if (!guild.allies_role || !guild.axis_role || !guild.spectator_role)
+        throw new BotError("Roles are not configured for this guild");
+
+    let role: string;
+    switch (user.team) {
+        case Team.ALLIES:
+            role = guild.allies_role;
+            break;
+        case Team.AXIS:
+            role = guild.axis_role;
+            break;
+        case Team.NONE:
+        case Team.AUTO:
+            role = guild.spectator_role;
+            break;
+    }
+
+    await member.roles
+        .set([
+            role,
+            ...[...member.roles.cache.keys()].filter(
+                (r) =>
+                    ![
+                        guild.allies_role,
+                        guild.axis_role,
+                        guild.spectator_role,
+                    ].includes(r)
+            ),
+        ])
+        .catch((e) => {
+            throw e;
+        });
+
+    setNickname(member, user);
+}
+
+export async function getGuild(guildId: string): Promise<GuildEntity> {
```

Comment:
shouldn't this be a repository method?

Reply:
we dont have a guild repository class, this is how we do it in the rest of the project (since its just a simple find)